common/lib/plugins/efm2/monitor_service.ts

```diff
@@ -0,0 +1,176 @@
+   *
+   * @param context The {@link MonitorConnectionContext} representing a connection.
+   * @param clientToAbort A reference to the connection associated with this context that will be aborted.
+   * */
+  stopMonitoring(context: MonitorConnectionContext, clientToAbort: any): Promise<void>;
+
+  releaseResources(): Promise<void>;
+}
+
+export class MonitorServiceImpl implements MonitorService {
+  private static readonly CACHE_CLEANUP_NANOS = BigInt(60_000_000_000);
+
+  protected static readonly monitors: SlidingExpirationCache<string, Monitor> = new SlidingExpirationCache(
+    MonitorServiceImpl.CACHE_CLEANUP_NANOS,
+    (monitor: Monitor) => monitor.canDispose(),
+    async (monitor: Monitor) => {
+      {
+        try {
+          await monitor.endMonitoringClient();
+        } catch (error) {
+          // ignore
+        }
+      }
+    }
+  );
+  private readonly pluginService: PluginService;
+  private telemetryFactory: TelemetryFactory;
+  private readonly abortedConnectionsCounter: TelemetryCounter;
+  monitorSupplier = (
+    pluginService: PluginService,
+    hostInfo: HostInfo,
+    properties: Map<string, any>,
+    failureDetectionTimeMillis: number,
+    failureDetectionIntervalMillis: number,
+    failureDetectionCount: number,
+    abortedConnectionsCounter: TelemetryCounter
+  ) =>
+    new MonitorImpl(
+      pluginService,
+      hostInfo,
+      properties,
+      failureDetectionTimeMillis,
+      failureDetectionIntervalMillis,
+      failureDetectionCount,
+      abortedConnectionsCounter
+    );
+
+  constructor(pluginService: PluginService) {
+    this.pluginService = pluginService;
+    this.telemetryFactory = pluginService.getTelemetryFactory();
+    this.abortedConnectionsCounter = this.telemetryFactory.createCounter("efm2.connections.aborted");
+  }
+
+  async startMonitoring(
+    clientToAbort: any,
+    hostInfo: HostInfo,
+    properties: Map<string, any>,
+    failureDetectionTimeMillis: number,
+    failureDetectionIntervalMillis: number,
+    failureDetectionCount: number
+  ): Promise<MonitorConnectionContext> {
+    const monitor = await this.getMonitor(hostInfo, properties, failureDetectionTimeMillis, failureDetectionIntervalMillis, failureDetectionCount);
+
+    if (monitor) {
+      const context = new MonitorConnectionContext(clientToAbort);
+      monitor.startMonitoring(context);
+      return context;
+    }
+
+    throw new AwsWrapperError(Messages.get("MonitorService.startMonitoringNullMonitor", hostInfo.host));
+  }
+
+  async stopMonitoring(context: MonitorConnectionContext, clientToAbort: ClientWrapper): Promise<void> {
+    if (context.shouldAbort()) {
+      context.setInactive();
+      try {
+        await clientToAbort.abort();
+        await clientToAbort.end();
+        this.abortedConnectionsCounter.inc();
+      } catch (error) {
+        // ignore
+        logger.debug(Messages.get("MonitorConnectionContext.exceptionAbortingConnection", error.message));
```

Comment:
Message should use error instead of exception 
```suggestion
        logger.debug(Messages.get("MonitorConnectionContext.errorAbortingConnection", error.message));
```